### DISCUSSION TITLE
xrepo 安装vcpkg::boost失败

### DISCUSSION THREAD

**Discussion by beholy** [xmake-io/xmake#3593] [General]
用xrepo安装vcpkg的boost包失败了,错误如下:
```
vcpkg install boost:x64-windows-static
Computing installation plan...
The following packages are already installed:
    boost[core]:x64-windows-static -> 1.81.0#3
boost:x64-windows-static is already installed
Restored 0 package(s) from C:\Users\xxx\AppData\Local\vcpkg\archives in 155 us. Use --debug to see more details.
Total install time: 882 us
checking for vcpkg::boost ... no

error: fetch vcpkg::boost-latest failed!
  => install vcpkg::boost latest .. failed
error: install failed!
error: execv(xmake require -v -j 8 --extra={system=false} vcpkg::boost) failed(-1)
```
vcpkg那边是安装成功了,但是xmake这边找不到,请问怎么处理?或者用xmake源里面的boost怎么编译带icu的locale库?谢谢

**Comment by waruqi** [comment]
boost 在 vcpkg 里包名应该是 boost[core]

add_requires("vcpkg::boost[core]")

**Reply by beholy** [reply]
谢谢回复,但是问题没能解决
```
Computing installation plan...
The following packages are already installed:
    boost[core]:x64-windows-static -> 1.81.0#3
boost:x64-windows-static is already installed
Restored 0 package(s) from C:\Users\xxx\AppData\Local\vcpkg\archives in 140 us. Use --debug to see more details.
Total install time: 888 us
checking for vcpkg::boost[core] ... no

error: fetch vcpkg::boost[core]-latest failed!
  => install vcpkg::boost[core] latest .. failed

fetch vcpkg::boost[core]-latest failed!
```
我使用 add_requires("vcpkg::implot", {alias = "implot"}) 是成功的,但boost不行.

这是vcpkg list的部分内容:
```
...
boost-variant2:x64-windows-static                 1.81.0#2            Boost variant2 module
boost-variant2:x86-windows                        1.81.0#2            Boost variant2 module
boost-variant:x64-windows-static                  1.81.0#2            Boost variant module
boost-variant:x86-windows                         1.81.0#2            Boost variant module
boost-vcpkg-helpers:x64-windows-static            1.81.0#3            Internal vcpkg port used to modularize Boost
boost-vcpkg-helpers:x86-windows                   1.81.0#3            Internal vcpkg port used to modularize Boost
boost-vmd:x64-windows-static                      1.81.0#2            Boost vmd module
boost-wave:x64-windows-static                     1.81.0#2            Boost wave module
boost-winapi:x64-windows-static                   1.81.0#2            Boost winapi module
boost-winapi:x86-windows                          1.81.0#2            Boost winapi module
boost-xpressive:x64-windows-static                1.81.0#2            Boost xpressive module
boost-yap:x64-windows-static                      1.81.0#2            Boost yap module
boost:x64-windows-static                          1.81.0#3            Peer-reviewed portable C++ source libraries
bzip2:x64-windows                                 1.0.8#3             bzip2 is a freely available, patent free, high-q
...
```
请您帮忙看下有没有办法解决?谢谢

**Comment by waruqi** [comment]
加 -vD 看下具体错误，然后到 https://github.com/xmake-io/xmake/blob/master/xmake/modules/package/manager/vcpkg/find_package.lua 里加点 print 自己调下

**Reply by beholy** [reply]
经过调试发现在find_package.lua中找不到pkgconfig files、includedirs、linkdirs and links

infofile指向的文件vcpkg\installed\vcpkg\info\boost_1.81.0_x64-windows-static.list内容如下：
```
x64-windows-static/
x64-windows-static/share/
x64-windows-static/share/boost/
x64-windows-static/share/boost/vcpkg.spdx.json
x64-windows-static/share/boost/vcpkg_abi_info.txt
```
而vcpkg\installed\vcpkg\info\boost-any_1.81.0_x64-windows-static.list内容是完整的
```
x64-windows-static/
x64-windows-static/include/
x64-windows-static/include/boost/
x64-windows-static/include/boost/any.hpp
x64-windows-static/include/boost/any/
x64-windows-static/include/boost/any/bad_any_cast.hpp
x64-windows-static/include/boost/any/basic_any.hpp
x64-windows-static/include/boost/any/fwd.hpp
x64-windows-static/include/boost/any/unique_any.hpp
x64-windows-static/share/
x64-windows-static/share/boost-any/
x64-windows-static/share/boost-any/copyright
x64-windows-static/share/boost-any/usage
x64-windows-static/share/boost-any/vcpkg.spdx.json
x64-windows-static/share/boost-any/vcpkg_abi_info.txt

```
您看需要怎么调整？是不是我安装导致的问题？

**Reply by waruqi** [reply]
> infofile指向的文件vcpkg\installed\vcpkg\info\boost_1.81.0_x64-windows-static.list内容如下：

里面的内容你也看到了，没任何 include ，那当然找不到了，至于为什么没有，这个你要问 vcpkg 了。。跟 xmake 没关系

> 而vcpkg\installed\vcpkg\info\boost-any_1.81.0_x64-windows-static.list内容是完整的

那是两个包，vcpkg 对 boost 不同组件做了拆分，不同 boost 子组件的包名都是不同的，`add_requires("vcpkg::boost[core]")` 只能找 boost 核心组件，any 组件得用 `add_requires("vcpkg::boost[any]")`

另外，xmake-repo 仓库原本就有 boost ，干嘛不用

**Reply by beholy** [reply]
谢谢回复,我开始用的是xmake-repo仓库的boost,但是链接不到locale库,就尝试用vcpkg,现在改回xmake-repo的boost,安装时加了configs,问题解决了

**Comment by smac89** [comment]
> > The content of the file vcpkg\installed\vcpkg\info\boost_1.81.0_x64-windows-static.list pointed to by infofile is as follows:
> 
> You have also seen the content inside. There is no include, so of course it cannot be found. As for why there is no include, you have to ask vcpkg. . It has nothing to do with xmake
> 
> > The content of vcpkg\installed\vcpkg\info\boost-any_1.81.0_x64-windows-static.list is complete
> 
> Those are two packages. vcpkg splits different components of boost. The package names of different boost sub-components are different. `add_requires("vcpkg::boost[core]")`You can only find boost core components, any components must be used `add_requires("vcpkg::boost[any]")`
> 
> In addition, the xmake-repo warehouse already has boost, why not use it?

Using that syntax does not work for me. For example, I'm trying to add `hello-imgui`, and when I tried using `vcpkg::hello-imgui[sdl2-binding]`, I get the following error message:

> $.dependencies[0].name (a package name): "hello-imgui[sdl2-binding]" is not a valid package name. Package names must be lowercase alphanumeric+hypens and not reserved (see https://learn.microsoft.com/vcpkg/users/manifests for more information)

**Comment by waruqi** [comment]
> > > The content of the file vcpkg\installed\vcpkg\info\boost_1.81.0_x64-windows-static.list pointed to by infofile is as follows:
> > 
> > 
> > You have also seen the content inside. There is no include, so of course it cannot be found. As for why there is no include, you have to ask vcpkg. . It has nothing to do with xmake
> > > The content of vcpkg\installed\vcpkg\info\boost-any_1.81.0_x64-windows-static.list is complete
> > 
> > 
> > Those are two packages. vcpkg splits different components of boost. The package names of different boost sub-components are different. `add_requires("vcpkg::boost[core]")`You can only find boost core components, any components must be used `add_requires("vcpkg::boost[any]")`
> > In addition, the xmake-repo warehouse already has boost, why not use it?
> 
> Using that syntax does not work for me. For example, I'm trying to add `hello-imgui`, and when I tried using `vcpkg::hello-imgui[sdl2-binding]`, I get the following error message:
> 
> > $.dependencies[0].name (a package name): "hello-imgui[sdl2-binding]" is not a valid package name. Package names must be lowercase alphanumeric+hypens and not reserved (see https://learn.microsoft.com/vcpkg/users/manifests for more information)

This is the vcpkg package name rule, xmake simply passes it to the vcpkg install command, if it is incorrect you can find the correct package name and feature configuration specification from the vcpkg documentation.

or you can run `vcpkg install hello-imgui[sdl2-binding]` command to debug and adjust it.
